### PR TITLE
[new release] minicaml (0.3)

### PR DESCRIPTION
**Pull request by 0x0f0f0f**
A simple, didactical, purely functional programming language

- Project page: <a href="https://github.com/0x0f0f0f/minicaml">https://github.com/0x0f0f0f/minicaml</a>
- Documentation: <a href="https://github.com/0x0f0f0f/minicaml">https://github.com/0x0f0f0f/minicaml</a>

### Added
- A simple AST optimizer
- Multiple let assignments (`let x = 2 and y = 3 and z = ... in ...`)
- Function currying (partial application)
- Strings
- Sequences of expressions separated with `;` are treated as a single expression
  that returns the value of the last one.
- Function pipelining (reverse composition) with the `>=>` operator.
  (`(fun x y -> x + y >=> g) 1 2` is `g 3`)
- Dictionaries (`{"hello": 123, "world": 345}`)
- Map, foldl and filter on lists and dictionaries
- Testing and code coverage using alcotest and bisect_ppx
### Bugfixes
- Fixed function application parser

### CODE REVIEW

File: packages/minicaml/minicaml.0.3/opam [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+]
+depends: [
+    "dune" {>= "2.0"}
+    "ocaml" {>= "4.05.0"}
+    "ANSITerminal"
+    "menhir"
+    "ppx_deriving"
+    "cmdliner"
+    "alcotest" {dev & >= "0.8.5"}
+    "bisect_ppx" {dev & >= "1.5.0"}
```
Comment (kit-ty-kate) [inline packages/minicaml/minicaml.0.3/opam]:
```suggestion
    "bisect_ppx" {with-test & >= "1.5.0"}
```

File: packages/minicaml/minicaml.0.3/opam [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  ["dune" "runtest" "-p" name] {dev}
+]
+depends: [
+    "dune" {>= "2.0"}
+    "ocaml" {>= "4.05.0"}
+    "ANSITerminal"
+    "menhir"
+    "ppx_deriving"
+    "cmdliner"
+    "alcotest" {dev & >= "0.8.5"}
```
Comment (kit-ty-kate) [inline packages/minicaml/minicaml.0.3/opam]:
```suggestion
    "alcotest" {with-test & >= "0.8.5"}
```

File: packages/minicaml/minicaml.0.3/opam [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+homepage: "https://github.com/0x0f0f0f/minicaml"
+bug-reports: "https://github.com/0x0f0f0f/minicaml/issues"
+dev-repo: "git+https://github.com/0x0f0f0f/minicaml.git"
+license: "MIT"
+synopsis: "A simple, didactical, purely functional programming language"
+description: "A simple, didactical, purely functional programming language written for the programming 2 course at the University of Pisa, extended with a simple parser made with Menhir and ocamllex"
+doc: "https://github.com/0x0f0f0f/minicaml"
+build: [
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name] {dev}
```
Comment (kit-ty-kate) [inline packages/minicaml/minicaml.0.3/opam]:
```suggestion
  ["dune" "runtest" "-p" name "-j" jobs] {with-test}
```
Test should run only when `with-test` is set (triggered by `opam install --with-test`)

File: packages/minicaml/minicaml.0.3/opam [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    "menhir"
+    "ppx_deriving"
+    "cmdliner"
+    "alcotest" {dev & >= "0.8.5"}
+    "bisect_ppx" {dev & >= "1.5.0"}
+]
+
+pin-depends: [
+  ["bisect_ppx.git" "git+https://github.com/aantron/bisect_ppx.git"]
+]
```
Comment (kit-ty-kate) [inline packages/minicaml/minicaml.0.3/opam]: `pin-depends` should not be part of the packages in opam-repository.

**Comment by 0x0f0f0f** [comment]
Fixing in next release. Thank you!

**Pull request closed without merging**